service/src/main/resources/db/migration/V388__preschool_assistance_decision_valid_to.sql

```diff
@@ -1,3 +1,24 @@
 ALTER TABLE assistance_need_preschool_decision
     ADD COLUMN valid_to DATE,
     ADD CONSTRAINT valid_from_before_valid_to CHECK (valid_from <= valid_to);
+
+WITH fill_in_valid_to AS (SELECT previous.id, min(next.valid_from) - interval '1 day' AS new_valid_to
+                          FROM assistance_need_preschool_decision previous,
+                               assistance_need_preschool_decision next
+                          WHERE previous.id != next.id
+                            AND previous.child_id = next.child_id
+                            AND previous.status = 'ACCEPTED'
+                            AND next.status = 'ACCEPTED'
+                            AND previous.valid_from < next.valid_from
+                            AND previous.valid_to IS NULL
+                          GROUP BY previous.id)
+UPDATE assistance_need_preschool_decision
+SET valid_to = fill_in_valid_to.new_valid_to
+FROM fill_in_valid_to
+WHERE fill_in_valid_to.id = assistance_need_preschool_decision.id;
+
+ALTER TABLE assistance_need_preschool_decision
+    ADD CONSTRAINT check$assistance_need_preschool_decision_no_overlap EXCLUDE USING gist (
+        child_id WITH =,
+        daterange(valid_from, valid_to, '[]') WITH &&
+        ) WHERE (status = 'ACCEPTED');
```

Comment:
Current model supports multiple accepted decisions starting on the same date so each city might need to check that they don't have such data in staging/prod or this will explode on deploy. Also might want to add some validation to check that in the controller, so that there's no http500 and alert on insert.

Reply:
> Also might want to add some validation to check that in the controller, so that there's no http500 and alert on insert.

Fixed